### PR TITLE
WIP:Bring back likes on comments

### PR DESCRIPTION
**Pull request by Flaburgan**
I started to work on that task. @denschub asked me how I was seeing the UI. Quite frankly, I thought about going with the exact same design than the like for posts in the stream. So, I move the timestamp of comment from the bottom of the comment to the top on the right of the author in #7905 and display a "like" link at the bottom of the comment. The avatar of the other likers will go there too, like for posts. This UI works well for the users at the moment so I didn't see need for changes, at least for now. This fits OK in the SPV too. I wanted to open this PR to have feedback about that before going too far, so please share your opinion :smile:

### CODE REVIEW

File: app/assets/javascripts/app/views/comment_view.js [hunk 1]
```diff
@@ -35,6 +36,11 @@ app.views.Comment = app.views.Content.extend({
 
   canRemove : function() {
     return app.currentUser.authenticated() && (this.ownComment() || this.postOwner());
+  },
+
+  toggleLike: function(evt) {
+    if(evt) { evt.preventDefault(); }
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_view.js]: Expected space(s) after "if".

File: app/services/comment_service.rb [hunk 2]
```diff
@@ -10,6 +10,11 @@ def create(post_id, text)
     user.comment!(post, text)
   end
 
+  def find!(comment_id)
+    #TODO check that the user has access to that comment first
```
Comment (diaspora-code-review) [inline app/services/comment_service.rb]: - Layout/LeadingCommentSpace: Missing space after `#`.
- Style/CommentAnnotation: Annotation keywords like `TODO` should be all upper case, followed by a colon, and a space, then a note describing the problem.

File: app/assets/javascripts/app/models/comment.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 // @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
 
 app.models.Comment = Backbone.Model.extend({
-  urlRoot: "/comments"
+  urlRoot: "/comments",
+
+  initialize: function() {
+    this.likeInteractions = new app.models.Post.LikeInteractions(_.extend({comment: this}, this.get("likeInteractions")));
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/comment.js]: Line 7 exceeds the maximum line length of 120.

File: app/assets/javascripts/app/models/post/interactions.js [hunk 4]
```diff
@@ -1,19 +1,15 @@
 // @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
 
 //require ../post
+//require like-interactions
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/interactions.js]: Expected space or tab after '//' in comment.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+// This class contains code extracted from interactions.js to factorize likes management between posts and comments
+app.models.Post.LikeInteractions = Backbone.Model.extend({
+  initialize : function(options){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: - Extra space after key 'initialize'.
- Missing space before opening brace.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    this.likes = new app.collections.Likes(this.get("likes"), options);
+  },
+
+  likesCount : function(){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: - Extra space after key 'likesCount'.
- Missing space before opening brace.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    return this.get("likes_count");
+  },
+
+  userLike : function(){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: - Extra space after key 'userLike'.
- Missing space before opening brace.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    return this.likes.select(function(like){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: Missing space before opening brace.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      return like.get("author") && like.get("author").guid === app.currentUser.get("guid");
+    })[0];
+  },
+
+  toggleLike : function() {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: Extra space after key 'toggleLike'.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    if(this.userLike()) {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: Expected space(s) after "if".

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+      this.unlike();
+    } else {
+      this.like();
+    }
+  },
+
+  like : function() {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: Extra space after key 'like'.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    var self = this;
+    this.likes.create({}, {
+      success: function() {
+        self.post.set({participation: true});
+        self.trigger("change");
+        self.set({"likes_count" : self.get("likes_count") + 1});
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: Extra space after key 'likes_count'.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+        self.likes.trigger("change");
+      },
+      error: function(model, response) {
+        app.flashMessages.handleAjaxError(response);
+      }
+    });
+  },
+
+  unlike : function() {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: Extra space after key 'unlike'.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+    var self = this;
+    this.userLike().destroy({
+      success: function() {
+        self.post.set({participation: false});
+        self.trigger('change');
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: Strings must use doublequote.

File: app/assets/javascripts/app/models/post/like_interaction.js [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+        self.set({"likes_count" : self.get("likes_count") - 1});
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interaction.js]: Extra space after key 'likes_count'.

File: app/assets/javascripts/app/views/comment_view.js [hunk 16]
```diff
@@ -6,10 +6,15 @@ app.views.Comment = app.views.Content.extend({
   className : "comment media",
   tooltipSelector: "time",
 
+  subviews : {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_view.js]: Extra space after key 'subviews'.

File: app/assets/javascripts/app/views/comment_view.js [hunk 17]
```diff
@@ -35,7 +40,16 @@ app.views.Comment = app.views.Content.extend({
...
     return app.currentUser.authenticated() && (this.ownComment() || this.postOwner());
-  }
+  },
+
+  toggleLike: function(evt) {
+    if (evt) { evt.preventDefault(); }
+    this.model.toggleLike();
+  },
+
+  likesInfoView : function(){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_view.js]: - Extra space after key 'likesInfoView'.
- Missing space before opening brace.

File: app/assets/javascripts/app/views/comment_view.js [hunk 18]
```diff
@@ -35,7 +40,16 @@ app.views.Comment = app.views.Content.extend({
...
+    return new app.views.LikesInfo({model : this.model});
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_view.js]: Extra space after key 'model'.

File: app/assets/javascripts/app/views/comment_view.js [hunk 19]
```diff
@@ -35,7 +40,16 @@ app.views.Comment = app.views.Content.extend({
...
+  },
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/comment_view.js]: Unexpected trailing comma.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 20]
```diff
@@ -0,0 +1,53 @@
+// This class contains code extracted from interactions.js to factorize likes management between posts and comments
+app.models.Post.LikeInteractions = Backbone.Model.extend({
+  initialize : function(options){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: - Extra space after key 'initialize'.
- Missing space before opening brace.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+    this.likes = new app.collections.Likes(this.get("likes"), options);
+  },
+
+  likesCount : function(){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: - Extra space after key 'likesCount'.
- Missing space before opening brace.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+    return this.get("likes_count");
+  },
+
+  userLike : function(){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: - Extra space after key 'userLike'.
- Missing space before opening brace.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+    return this.likes.select(function(like){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: Missing space before opening brace.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 24]
```diff
@@ -0,0 +1,53 @@
...
+      return like.get("author") && like.get("author").guid === app.currentUser.get("guid");
+    })[0];
+  },
+
+  toggleLike : function() {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: Extra space after key 'toggleLike'.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+    if(this.userLike()) {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: Expected space(s) after "if".

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 26]
```diff
@@ -0,0 +1,53 @@
...
+      this.unlike();
+    } else {
+      this.like();
+    }
+  },
+
+  like : function() {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: Extra space after key 'like'.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 27]
```diff
@@ -0,0 +1,53 @@
...
+    var self = this;
+    this.likes.create({}, {
+      success: function() {
+        self.post.set({participation: true});
+        self.trigger("change");
+        self.set({"likes_count" : self.get("likes_count") + 1});
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: Extra space after key 'likes_count'.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 28]
```diff
@@ -0,0 +1,53 @@
...
+        self.likes.trigger("change");
+      },
+      error: function(model, response) {
+        app.flashMessages.handleAjaxError(response);
+      }
+    });
+  },
+
+  unlike : function() {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: Extra space after key 'unlike'.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 29]
```diff
@@ -0,0 +1,53 @@
...
+    var self = this;
+    this.userLike().destroy({
+      success: function() {
+        self.post.set({participation: false});
+        self.trigger('change');
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: Strings must use doublequote.

File: app/assets/javascripts/app/models/post/like_interactions.js [hunk 30]
```diff
@@ -0,0 +1,53 @@
...
+        self.set({"likes_count" : self.get("likes_count") - 1});
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/like_interactions.js]: Extra space after key 'likes_count'.

File: app/assets/javascripts/app/models/post/interactions.js [hunk 31]
```diff
@@ -1,19 +1,15 @@
 // @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
 
-//require ../post
+//= require ./like_interactions
 
-app.models.Post.Interactions = Backbone.Model.extend({
+app.models.Post.Interactions = app.models.Post.LikeInteractions.extend({
   initialize : function(options){
+    app.models.Post.LikeInteractions.prototype.initialize({post : this.post}); // I don't know how to call the super method yet
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/interactions.js]: - Line 7 exceeds the maximum line length of 120.
- Extra space after key 'post'.

File: app/assets/javascripts/app/models/post/interactions.js [hunk 32]
```diff
@@ -1,75 +1,29 @@
 // @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
 
-//require ../post
+//= require ./like_interactions
 
-app.models.Post.Interactions = Backbone.Model.extend({
-  initialize : function(options){
+app.models.Post.Interactions = app.models.Post.LikeInteractions.extend({
+  initialize: function(options) {
+    app.models.Post.LikeInteractions.prototype.initialize({post: this.post}); // I don't know how to call the super method yet
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/models/post/interactions.js]: Line 7 exceeds the maximum line length of 120.

**Comment by LorenzoAncora** [comment]
> I wanted to open this PR to have feedback about that before going too far, so please share your opinion [smile]

@Flaburgan _I appreciate your commitment but I don't like the idea_ of changing the UX so drastically.<br/>I don't like this feature because it is **addictive** and I think it's totally **superfluous**, if not harmful.
**I prefer that my interlocutors do not limit their opinion to a "like", especially if they have not participated in the discussion.**
While the likes on a topic are a convenience, **the likes on individual comments will transform the social network into a competition to take more likes, generate anxiety and favor ad personam attacks and the creation of sockpuppets** (as often happens on Facebook<sup>&reg;</sup>, _people think "I have more likes and so I'm right / I'm better than you / I'm more popular than you"_). This worsens the quality of the discussions and could push part of the user towards other social networks.
If your goal is not to create a Facebook<sup>&reg;</sup> clone, I would say that you should avoid it or **make it an opt-in** (that is, whoever publishes the comment must explicitly enable this feature).
Have a good time.

**Comment by Flaburgan** [comment]
Hi @LorenzoAncora and thank you for taking time to share you thoughts about diaspora*. These worries about users using only likes instead of writing actual content are something often brought since I started working on reviving likes on comment. Despite this risk exists, I don't think this is going to happen. Likes mainly mean "I agree". It is a way to give importance to a comment. This can be achieve by posting another comment saying exactly that, but this doesn't work well at the moment because diaspora* doesn't have (yet?) threaded comments. That means if someone comments saying something you agree, but you arrive 1 hour after the comment has been written and 20 others comments have been posted since then, possibly driving the discussion in another direction, then writing another comment at the very bottom saying "I agree with comment 1" is not nice. So many users block themselves and don't do that. So instead of removing interaction because of less complete like instead of detailed comment, I think likes on comment will bring more interactions than before, make diaspora* a more alive place. 

So we're going to add likes on comment. If we do see some "bad" behaviors coming from that, we could remove the feature again. Future will tell us. And btw, if you **really don't** want to see likes on comment, just remove that commit from your pod. That's the power of free software ;)

**Comment by LorenzoAncora** [comment]
> [...] **if you really don't want to see likes on comment, just remove that commit from your pod** [...]

@Flaburgan , **this only applies if each individual user has his own pod, which is of course false**; furthermore, **users will still be able to put likes on another pod**, so removing the commit from a single pod is totally useless.
The functionality is not an opt-i, when it could be. :-(

> [...] Despite this risk exists, I don't think this is going to happen. [...] likes on comment will bring more interactions than before, make diaspora* a more alive place [...]

The likes on individual comments **will transform the social network into a competition to take more likes, generate anxiety, personal attacks and the creation of (even more) sockpuppets**. 
It already happens on other social networks and this favors a toxic climate, why should Diaspora be an exception? I consider it a malicious feature that makes it unpleasant to be on the social network and I think it should be an opt-in. We already have likes on posts, they are enough.
**Only accounts that enable the feature should be able to receive this kind of votes on their comments, I don't understand why you have to force all existing users. Don't copy Facebook, make the feature optional, so everyone will be happy.**

> [...] we're going to add likes on comment [...]

What is the point of asking for opinions and then ignoring them?
**_Please_**, just make the possibility of receiving "like" an opt-in for those who create the comment, it's about adding a single switch to the user profile. If the end user does not enable it, no pod can send like to the comment produced by that user.
I know that you care so much to add this source code (and I appreciate), but please don't impose yourself on users, at least make it optional. You can consider this message as a feature-request for the next release. :-)

**Comment by denschub** [comment]
This is a pull request, and as such, is a place to discuss nuances about the technical side of the code submitted here. This is not the place to have philosophical discussions, and this is not the place to question the opinion of a majority of users. If anything, this place would be [here](https://discourse.diasporafoundation.org/t/bring-back-likeable-comments/112).

**Comment by LorenzoAncora** [comment]
> This is a pull request, and as such, is a place to discuss nuances about the technical side of the code submitted here.

### @denschub just deleted my polite answer and requests to @Flaburgan, disrupting our discussion.
**I did not speak of philosophy but only of the end user experience and the negative impact of this functionality on it.** It is not the first time that he has censored a user, but only because his/her opinion does not please him. This is not the way to manage a community, this is not the way to be respected.
**It makes no sense to ask users for their opinion if you delete them (or at least delete only those that you don't like, even if they are polite). Honestly I feel offended by this behavior, because I have never allowed myself to censor the opinion of users, even when I didn't like them.** I'm offended and I'm leaving, goodbye.

---

@Flaburgan the important thing is that at least you have read it, even if it's worth zero (if you haven't, check your mailbox for a copy). I don't know if anything else has been canceled, however I am on Diaspora where (_for the moment_) we can discuss in peace. On my profile I don't censor anyone's opinion, because all opinions have the same importance and the same respectability for me, so it's a better place to discuss.

**Comment by tclaus** [comment]
@Flaburgan @denschub Any Progress here? 
Will this be merged for the 8?

**Comment by Flaburgan** [comment]
@tclaus unfortunately I switched my main focus of contribution to another project. I don't think I will continue to work on this. Feel free to follow up if you want to.

**Comment by tclaus** [comment]
> @tclaus unfortunately I switched my main focus of contribution to another project. I don't think I will continue to work on this. Feel free to follow up if you want to.

Is there anything left over I can help with?

**Comment by Flaburgan** [comment]
My current work is pushed here. Honestly I don't remember exactly where I stopped, it was 2 years ago...

**Comment by denschub** [comment]
I don't think I'm blocking this PR, but just in case

>  @denschub asked me how I was seeing the UI [...] I wanted to open this PR to have feedback about that before going too far, so please share your opinion 😄

is a direct request to me: I really have no opinion here. This feature is one of those things that I don't *like* (pun intended), and if it would be completely up to me, we'd have no "likes" at all, not even for posts. Luckily, it's not up to me, and since people really want this, I'll not get in anyone's way.

**Comment by tclaus** [comment]
See the like? 😉
So let me see if I can contribute something to close this issue.

**Comment by denschub** [comment]
Obsoleted by #8203

**Pull request closed without merging**